### PR TITLE
Add AssemblyAI transcript API as loaders

### PR DESCRIPTION
**Pull request by Swimburger**
This PR adds multiple document loaders for loading a transcript from a local or online audio or video file, by using [the AssemblyAI transcript API](https://www.assemblyai.com/docs/API%20reference/transcript).

At its simplest, you can use the `AudioTranscriptLoader` to get a transcript back from an audio file using TS/JS like this:

```typescript
import {
  AudioTranscriptLoader,
  AudioTranscriptParagraphsLoader,
  AudioTranscriptSentencesLoader
} from "langchain/document_loaders/web/assemblyai";

// You can also use a local file path and the loader will upload it to AssemblyAI for you
const audioUrl = "https://storage.googleapis.com/aai-docs-samples/espn.m4a";

// use `AudioTranscriptParagraphsLoader` or `AudioTranscriptSentencesLoader` for splitting the transcript into paragraphs or sentences
const loader = new AudioTranscriptLoader(
  {
    audio_url: audioUrl
    // any other parameters as documented here: https://www.assemblyai.com/docs/API%20reference/transcript#create-a-transcript 
  });
const docs = await loader.load();
console.dir(docs, { depth: Infinity });
```

This is my first time contributing to LangChain, so any feedback is very welcome! 
In the future, we plan to contribute some tools for working with transcripts and a Python version as well.

Twitter handles to shout out if so kindly 🙇 
[@AssemblyAI](https://twitter.com/AssemblyAI) and [@RealSwimburger](https://twitter.com/RealSwimburger)

### CODE REVIEW

File: langchain/src/document_loaders/tests/assemblyai.int.test.ts [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+test("Test Audio Transcript Loader", async () => {
+  const loader = new AudioTranscriptLoader({
+    audio_url: "https://storage.googleapis.com/aai-docs-samples/nbc.mp3",
+  });
+  const docs = await loader.load();
+
+  expect(docs).toHaveLength(1);
+  expect(docs[0].pageContent).not.toBeFalsy();
+  expect(docs[0].metadata).not.toBeFalsy();
+  console.dir(docs, { depth: Infinity });
```
[the remark this answers is not in the document]
Comment (Swimburger) [reply langchain/src/document_loaders/tests/assemblyai.int.test.ts:31]: I didn't do it intentionally, I saw other integration tests doing this and kept it.

File: docs/extras/modules/data_connection/document_loaders/integrations/web_loaders/assemblyai_audio_transcription.mdx [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+hide_table_of_contents: true
+---
+
+# AssemblyAI Audio Transcript 
+
+This covers how to load audio (and video) transcripts as document objects from a file using the [AssemblyAI API](https://www.assemblyai.com/docs/API%20reference/transcript).
+
+## Usage
+
+To use the loaders you need an [Assembly AI account](https://www.assemblyai.com/dashboard/signup) and
```
[the remark this answers is not in the document]
Comment (Swimburger) [reply docs/extras/modules/data_connection/document_loaders/integrations/web_loaders/assemblyai_audio_transcription.mdx:11]: Absolutely, my bad.

File: langchain/src/document_loaders/web/assemblyai.ts [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+   * and loads the paragraphs of the transcript, creating a document for each paragraph.
+   * @returns A promise that resolves to an array of documents, each containing a paragraph of the transcript.
+   */
+  override async load(): Promise<Document<TranscriptSegment>[]> {
+    let transcript = await this.client.createTranscript(
+      this.createTranscriptParams
+    );
+    transcript = await this.client.waitForTranscriptToComplete(transcript.id);
+    const paragraphsResponse = await this.client.getParagraphs(transcript.id);
+    console.dir(paragraphsResponse);
```
[the remark this answers is not in the document]
Comment (Swimburger) [reply langchain/src/document_loaders/web/assemblyai.ts]: Nope, well spotted.

File: langchain/src/document_loaders/web/assemblyai.ts [hunk 4]
```diff
@@ -0,0 +1,211 @@
...
+  /**
+   * Creates a new AudioSubtitleLoader.
+   * @param createTranscriptParams The parameters to create the transcript.
+   * @param subtitleFormat The format of the subtitles, either `srt` or `vtt`.
+   * @param assemblyAIOptions The options to configure the AssemblyAI loader.
+   * Configure the `assemblyAIOptions.apiKey` with your AssemblyAI API key, or configure it as the `ASSEMBLYAI_API_KEY` environment variable.
+   */
+  constructor(
+    private createTranscriptParams: CreateTranscriptParams,
+    private subtitleFormat: "srt" | "vtt" = "srt",
```
[the remark this answers is not in the document]
Comment (Swimburger) [reply langchain/src/document_loaders/web/assemblyai.ts]: I made it into an enum, but Rollup seems to have a bug with enums, so it's a const obj with the same UX as enums.

File: langchain/package.json [hunk 5]
```diff
@@ -529,7 +532,7 @@
     "release": "release-it --only-version --config .release-it.json",
     "test": "NODE_OPTIONS=--experimental-vm-modules jest --testPathIgnorePatterns=\\.int\\.test.ts --testTimeout 30000",
     "test:watch": "NODE_OPTIONS=--experimental-vm-modules jest --watch --testPathIgnorePatterns=\\.int\\.test.ts",
-    "test:integration": "NODE_OPTIONS=--experimental-vm-modules jest --testPathPattern=\\.int\\.test.ts --testTimeout 100000",
+    "test:integration": "NODE_OPTIONS=--experimental-vm-modules jest --testPathPattern=assemblyai\\.int\\.test.ts --testTimeout 100000",
```
Comment (jacoblee93) [inline langchain/package.json:535]: You can just run `yarn test:single path/to/test`!

Comment (Swimburger) [reply langchain/package.json:535]: My bad, it was hard to get the testing stuff to work. Thanks for the heads up!

**Comment by jacoblee93** [comment]
@Swimburger pushed a few build fixes and moved the inlined code into imported examples - it's good with me but want to give it a final look?

Will aim to merge tonight and ship a release! Thank you.

**Pull request merged**